core/blockchain/abci/abci.go

```diff
@@ -194,6 +195,161 @@ func AddCommonDeliverTxEvents(resp types.ResponseDeliverTx, tx Tx) types.Respons
 	return resp
 }
 
+const (
+	TxTypeStakeDeposited                 = "StakeDeposited"
+	TxTypeStakeRemoved                   = "StakeRemoved"
+	TxTypeBuiltinAssetDeposit            = "BuiltinAssetDeposit"
+	TxTypeBuiltinAssetWithdrawal         = "BuiltinAssetWithdrawal"
+	TxTypeERC20Deposit                   = "ERC20Deposit"
+	TxTypeERC20Withdrawal                = "ERC20Withdrawal"
+	TxTypeTransfer                       = "Transfer"
+	TxTypeOrderSubmission                = "OrderSubmission"
+	TxTypeOrderCancellation              = "OrderCancellation"
+	TxTypeOrderAmendment                 = "OrderAmendment"
+	TxTypeVoteSubmission                 = "VoteSubmission"
+	TxTypeWithdrawSubmission             = "WithdrawSubmission"
+	TxTypeLiquidityProvisionSubmission   = "LiquidityProvisionSubmission"
+	TxTypeLiquidityProvisionCancellation = "LiquidityProvisionCancellation"
+	TxTypeProposalSubmission             = "ProposalSubmission"
+	TxTypeAnnounceNode                   = "AnnounceNode"
+	TxTypeNodeVote                       = "NodeVote"
+	TxTypeNodeSignature                  = "NodeSignature"
+	TxTypeOracleDataSubmission           = "OracleDataSubmission"
+	TxTypeDelegateSubmission             = "DelegateSubmission"
+	TxTypeUndelegateSubmission           = "UndelegateSubmission"
+	TxTypeKeyRotateSubmission            = "KeyRotateSubmission"
+	TxTypeStateVariableProposal          = "StateVariableProposal"
+	TxTypeCancelTransfer                 = "CancelTransfer"
+	// TxTypeValidatorHeartbeat                 = "ValidatorHeartbeat".
```

Comment:
why is this excluded?

Reply:
@jeremyletang agreed. There was a discussion on if heartbeats should be included. We all agreed they should, because *excluding* them was misleading.

Reply:
Are we sure we want to add the attributes for all the heartbeats to the attributes table? I mean, fine by me